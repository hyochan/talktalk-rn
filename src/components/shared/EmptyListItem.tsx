import React, { Component } from 'react';
import {
  ActivityIndicator,
  StyleSheet,
  Image,
  Text,
  View,
  ViewStyle,
} from 'react-native';

interface IStyles {
  wrapper: ViewStyle;
}

const styles: IStyles = StyleSheet.create({
  wrapper: {
    flex: 1,
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'stretch',
  },
});

interface IProps {
  style: ViewStyle;
  children?: any;
}

function Shared(props: IProps) {
  return (
    <View
      style={props.style}
    >
      <Text
        style={{
          fontSize: 14,
          color: 'rgb(155,155,155)',
          alignSelf: 'center',
        }}
      >{props.children}</Text>
    </View>
  );
}

Shared.defaultProps = {
  style: styles.wrapper,
};

export default Shared;
