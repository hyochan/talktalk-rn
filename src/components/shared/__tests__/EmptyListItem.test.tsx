import 'react-native';
import * as React from 'react';
import EmptyListItem from '../EmptyListItem';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-native-testing-library';

const component: React.ReactElement = <EmptyListItem/>;

describe('[EmptyListItem] rendering test', () => {
  it('renders as expected', () => {
    const disabledComponent: React.ReactElement = <EmptyListItem/>;
    const json = renderer.create(disabledComponent).toJSON();
    expect(json).toMatchSnapshot();
  });
});
