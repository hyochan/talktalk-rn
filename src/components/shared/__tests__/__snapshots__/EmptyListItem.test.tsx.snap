// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[EmptyListItem] rendering test renders as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "rgb(155,155,155)",
        "fontSize": 14,
      }
    }
  />
</View>
`;
